core/src/com/unciv/logic/automation/unit/WorkerAutomation.kt

```diff
@@ -138,14 +138,20 @@ class WorkerAutomation(
 
             // If there's move still left, perform action
             // Unit may stop due to Enemy Unit within walking range during doAction() call
-            if (unit.currentMovement > 0 && reachedTile == tileToWork) {
```

Comment:
Why did you remove this? This is a necessary line, workers should not start removing things if they have 0 movement left this turn